src/installer/_core.py

```diff
@@ -0,0 +1,113 @@
+"""Wheel Installation logic, sans I/O
+"""
+
+import posixpath
+
+from installer._compat import StringIO
+from installer._compat.typing import TYPE_CHECKING
+from installer.parsers import parse_metadata_file
+from installer.protocols import Destination, Scheme, WheelSource
+from installer.validation import InvalidWheelSource
+
+if TYPE_CHECKING:
+    from typing import Callable, List, Optional
+
+    from installer._compat.typing import Text
+
+    Validator = Callable[[WheelSource], None]
+
+
+__all__ = ["Installer"]
+
+
+class Installer(object):
+    """Implements wheel installation logic
+
+    Supports Wheel version 1.0 (PEP 427).
+    """
+
+    def __init__(self, name, validators):
+        # type: (Optional[Text], List[Validator]) -> None
+        super(Installer, self).__init__()
+        self.name = name
+        self._validators = validators
+
+    def install(self, source, destination, requested=False):
+        # type: (WheelSource, Destination, bool) -> None
+        """Install a wheel, as described by ``source``, to ``destination``.
+        """
+
+        # Process the WHEEL file
+        wheel_metadata_stream = source.open_dist_info(u"WHEEL")
```

Comment:
I felt that each step is really trivial and making them into private functions felt like an unnecessary strong tool.

The only thing I that thought could be useful is splitting the function into the "check WHEEL stuff" and "spread things out stuff". And, each of those is individually simple enough that it felt unnecessary to break them out into separate functions.

FWIW, I think the comments do a good enough job of describing what's happening, and the code flow is straightforward enough to follow.

We can definitely break this up if things get difficult to follow, but we're far from being in that position here. Maybe, adding RecordKeeper might be the reason to justify make the switch, since it'll add a bunch more complexity to the "spread stuff" part of this function.

(see https://github.com/pypa/pip/blob/master/src/pip/_internal/operations/prepare.py#L379 for what I think of as too-big-need-to-break).

Reply:
I always feel that when I need comments to tell what's happening and not why as a code smell and the architecture could be improved.

Reply:
I don't think I've ever thought of comments as being for describing "why" vs "how". Some food for thought.

FWIW, I'll be breaking this up when I add the record handling logic back into this class, so your point is well received.